ionic_user_interface/src/components/UserRouteList.vue

```diff
@@ -0,0 +1,141 @@
+<template>
+  <ion-list class="ion-no-margin ion-no-padding">
+    <ion-card
+      v-for="(route, index) of routes"
+      :key="index"
+      class="ion-text-left route-card"
+      @click="() => handleRouteCardClick(route.username, route.createdAt)"
+    >
+      <ion-card-header>
+        <ion-card-title>{{ route.routeName }}</ion-card-title>
+        <VoteButton
+          class="vote-button"
+          :username="route.username"
+          :createdAt="route.createdAt"
+          v-model:voteCount="route.voteCount"
+          v-model:hasVoted="route.hasVoted"
+        ></VoteButton>
+      </ion-card-header>
+      <ion-card-content>
+        <b>Gym:&nbsp;</b>
+        <router-link
+          :to="'/gym/' + route.gymLocation + '/' + route.gymName"
+          @click.capture.stop="undefined"
+        >
+          {{ route.gymName }}
+        </router-link>
+        <br />
+        <b>Grade:</b>
+        V{{ route.publicGrade }}
+        <br />
+        <b>Created:</b>
+        {{ route.createdAt.split('T')[0] }}
+      </ion-card-content>
+    </ion-card>
+  </ion-list>
+</template>
+
+<script lang="ts">
+import { defineComponent, inject, Ref, ref, onMounted } from 'vue';
+import { heart, heartOutline } from 'ionicons/icons';
+import { IonCard, IonCardContent, IonCardHeader, IonCardTitle, IonList } from '@ionic/vue';
+import { useRouter, useRoute } from 'vue-router';
+import axios from 'axios';
+
+import VoteButton from '@/components/VoteButton.vue';
+import { throttle } from 'lodash';
+
+interface Route {
```

Comment:
Maybe we should extract this out somewhere into an interfaces.d.ts or smth

Reply:
Different types for `GymRoute` and `UserRoute` tho

Reply:
ok then leave it